lib/tenstorrent/fwupdate/fwupdate.c

```diff
@@ -186,43 +167,32 @@ int tt_fwupdate(const char *tag, bool dry_run, bool reboot)
 	tt_boot_fs_fd fd;
 	bool found = false;
 
-	rc = tt_boot_fs_mount(&boot_fs, z_tt_boot_fs_read, z_tt_boot_fs_write, z_tt_boot_fs_erase);
-	if (rc != TT_BOOT_FS_OK) {
-		LOG_DBG("%s() failed: %d", "tt_boot_fs_mount", rc);
-		return -EIO;
-	}
-
-	if (!device_is_ready(flash1_dev)) {
-		LOG_DBG("Device %s is not ready", flash1_dev->name);
+	if (!device_is_ready(tt_boot_fs.dev)) {
+		LOG_DBG("Device %s is not ready", tt_boot_fs.dev->name);
 		return -ENODEV;
 	}
 
-	LOG_DBG("Parsing SPI flash %s", flash1_dev->name);
+	LOG_DBG("Parsing SPI flash %s", tt_boot_fs.dev->name);
 
-	for (uint32_t addr = TT_BOOT_FS_OFFSET + TT_BOOT_FS_FD_HEAD_ADDR, i = 0;
-	     i < CONFIG_TT_BOOT_FS_IMAGE_COUNT_MAX; addr = tt_boot_fs_next(addr), ++i) {
+	tt_boot_fs_fd fds[32];
```

Comment:
Would be best to use a common Kconfig value here.

E.g. `tt_boot_fs_fd fds[CONFIG_TT_BOOT_FS_MAX_FDS];`